platform/src/test/java/org/exoplatform/platform/qa/ui/platform/wiki/functional/WikiTemplateTestIT.java

```diff
@@ -0,0 +1,732 @@
+   * Settings - Select Template tab - Click on Add More Templateslink Expected
+   * Outcome: Form to create new template appears Step number: 2 Step Name: - Step
+   * Description: Step 2: Add new template Input Data: - Put title, description,
+   * content - Click on Save Template Expected Outcome: New template is created.
+   * It'll be shown in Template form Step number: 3 Step Name: - Step Description:
+   * Step 3: Edit template Input Data: - Click Edit icon corresponding with the
+   * template want to edit - Change values - Click on Save Template Expected
+   * Outcome: Template is edited successful
+   */
+  @Test
+  public void test05_EditCreatedTemplate() {
+    info("Test 5: Edit created template");
+
+    info("Go to Wiki Settings");
+    homePagePlatform.goToWiki();
+    wikiHomePage.goToWikiSettingPage();
+    info("Add a new template");
+    String title = "title" + getRandomNumber();
+    String des = "des" + getRandomNumber();
+    String content = "content" + getRandomNumber();
+    wikiSettingPage.addTemplate(title, des, content);
+    wikiSettingPage.saveTemplate();
+    wikiHomePage.confirmWaringMessage(true);
+    info("Verify that new tempate is created. It'll be shown in template form");
+    wikiValidattions.verifyTemplateInList(title);
+    info("Edit a template");
+    String newTitle = "newTitle" + getRandomNumber();
+    String newDes = "newDes" + getRandomNumber();
+    String newContent = "newContent" + getRandomNumber();
+    wikiSettingPage.editTemplate(title, newTitle, newDes, newContent);
+    info("Verify that new tempate is created. It'll be shown in template form");
+    wikiValidattions.verifyTemplateInList(newTitle);
+    wikiSettingPage.deleteTemplate(newTitle);
+  }
+
+  /**
+   * <li>Case ID:139402.</li>
+   * <li>Test Case Name: Search template when Search field is blank.</li>
+   * <li>Pre-Condition:</li>
+   * <li>Post-Condition:</li> Step Number: 1 Step Name: - Step Description: Step
+   * 1: Open form to create new template Input Data: - Go to Browser -> Wiki
+   * Settings - Select Template - Click on Add More Templateslink Expected
+   * Outcome: Form to create new template appears* Step number: 2 Step Name: -
+   * Step Description: Step 2: Add new template Input Data: - Put title,
+   * description, content - Click on Save Template Expected Outcome: New template
+   * is created. It'll be shown in Template form* Step number: 3 Step Name: - Step
+   * Description: Step 3: Search template whenSearch field is blank Input Data: -
+   * Leave Search field is blank - Press Enter key Expected Outcome: All templates
+   * are listed
+   */
+  @Test
+  public void test06_SearchTemplateWhenSearchFieldIsBlank() {
+    info("Test 6: Search template when Search field is blank");
+
+    info("Go to Wiki Settings");
+    homePagePlatform.goToWiki();
+    wikiHomePage.goToWikiSettingPage();
+    ArrayList<String> arrayTemplate = new ArrayList<String>();
```

Comment:
Always use the interface when possible, here it should be : List<String> arrayTemplate = new ArrayList<String>();

Reply:
done